app/services/katello/upstream_connection_checker.rb

```diff
@@ -22,7 +21,6 @@ def can_connect?
     end
 
     def assert_connection
-      assert_connected
```

Comment:
Looking at the Ruby changes as a whole, is it really valid to remove all of this? We relied on this to check if we're in disconnected mode via the #ping method in UpstreamSubscriptionsController

Reply:
SubscriptionsPage is the only consumer for the disconnected mode and won't care about it any more after this PR. 
Is the method still necessary if no one cares about  it?

Reply:
I think it needs to stay because what changed is:

1. removed call to /api/v2/settings/content_disconnected from the UI to determine if we are in disconnected mode
2. rely totally on /api/v2/upstream_subscriptions/ping to determine if we are in disconnected mode from UI perspective

This means that removing this code breaks the very solution that this PR presents, because now there is *no* way to know if we are disconnected mode in the subscriptions UI

Reply:
I get your point and think you are right.
My concern is `assert_connected` does not work from non-admin users.

Reply:
I think it will work. The reason is that on the backend side - we aren't filtering any of that based on permissions, and also the disconnected setting check is not using the API. Since we're on the backend side it's just looking in the DB!

Reply:
Of course, better to verify that than take my work for it ;)